vscode-lean4/src/leanclient.ts

```diff
@@ -409,6 +430,7 @@ export class LeanClient implements Disposable {
         assert(() => this.isStarted())
         if (this.client && this.running) {
             this.noPrompt = true;
+            this.showingRestartMessage = true
```

Comment:
modifying `this.showingRestartMessage` doesn't belong here, it should be localized to  the showRestartMessage method.